### PR TITLE
Support array type in go generate with whitelist

### PR DESCRIPTION
**Pull request by olibaa**
Fixes: https://github.com/google/go-github/issues/2425

This PR is similar to the following PR:
- https://github.com/google/go-github/pull/1867

### CODE REVIEW

File: github/gen-accessors.go
Comment (gmlewis) [inline github/gen-accessors.go:373]:
```suggestion
// Get{{.FieldName}} returns the {{.FieldName}} {{if .MapType}}map{{else if .ArrayType }}slice{{end}} if it's non-nil, {{if .MapType}}an empty map{{else if .ArrayType }}nil{{end}} otherwise.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
While I appreciate what you are trying to do, @olibaa, this is not how I think it should be done.

Please read very carefully what I wrote [here](https://github.com/google/go-github/issues/2425#issuecomment-1200035340). 
I do not want to add getters for every possible slice.
They are simply unnecessary and add a huge amount of bloat for zero benefit.

What I recommend here is that you add a whitelist of fields that your code should process.
So, for example, we have only a single field currently that we want to add to the whitelist: `PushEvent.Commits`.

Then, on lines 140-142 of github/gen-accessors.go, you would check this whitelist and only call `t.addArrayType` if the current field matches an item in the whitelist (which itself could simply be a `var whitelistSliceGetters = map[string]bool{ ...: true}` to make lookup super-easy).

That way, when we look at the diffs of github/github-accessors.go and github/github-accessors_test.go, we should only see one new field in this PR... the one for `PushEvent.Commits`.

How does that sound to you?

**Comment by olibaa** [comment]
@gmlewis
Sounds good!
I will fix the code.

File: github/gen-accessors.go
Comment (olibaa) [inline github/gen-accessors.go:140]: @gmlewis 
I add my whitelist here

Is my understanding correct?

Comment (gmlewis) [reply github/gen-accessors.go:140]: Yes, you check your whitelist here. If it matches, then you go ahead with the operation, otherwise you ignore it.

Comment (olibaa) [reply github/gen-accessors.go:140]: thanks for answering !

Comment (olibaa) [reply github/gen-accessors.go:145]: @gmlewis 
Fixed!
I would like a review 🙇‍♂️

**Review by gmlewis — APPROVED** [review]
Beautiful, @olibaa!  Well done!
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
